### DISCUSSION TITLE
Shipping the theme downstream

### DISCUSSION THREAD

**Discussion by Chrysostomus** [rafaelmardojai/firefox-gnome-theme#528] [General]
Hi,

I was notified of this post
https://floss.social/@rafaelmardojai/109532714277650579

I could not respond directly, as my mastadon account is still pending review. So I opened this issues since I have no other means of direct communication. Sorry for polluting your issue tracker.

Manjaro has shipped the upstream version of this theme since October. We no longer ship the modified version. The current implementation is that our greeter application has a toggle that installs and uninstalls the theme using the script in this repository. If you find our using of your theme disagreeable, we can also not ship it. I admire your work, and despite any reputation Manjaro might have on the regard, we do try to respect the upstream developers work and wishes. In sorry if I've inconvenienced you unduly with this.

**Comment by rafaelmardojai** [comment]
Hi,

The way you describe how this is implemented on Manjaro now make me think that is OK, but the reddit post linked in mastodon demonstrate that some users aren't getting this right away. Maybe you can improve somehow the way this setting is exposed, maybe add some warning explaining the possible problems. Is this toggle exposed in some initial setup?

I really don't feel comfortable with users complaining to Firefox when they are using my theme, that's all I want to avoid. I have even considered many times to stop this theme development because of that.

I understand that you could want to ease the access to this more "GNOMISH" experience to your users, but I don't think it should be too easily accessible, we're monkey patching Firefox and many times I don't have enough time to fix issues at time. Many times we just see small broken parts, but this time the UI was mostly unusable.

I hope we can get to something that makes all parties comfortable, me, Manjaro and Firefox (it's well know the warning message  you get when you post something manjaro related on firefox's reddit)

**Comment by emansom** [comment]
Please hide this under an advanced configurable toggle hidden in an advanced menu somewhere, with off being the default.

Send a notification on each GNOME shell session start that it's enabled as well.

Manjaro is being pushed by YouTubers to beginners, and should therefore assume not everyone in its user base is knowledgeable enough to debug issues themselves.

**Comment by Chrysostomus** [comment]
We can disable it by default and relabel the toggle for the theme. It needs some extra explanation anyway, because it has been unclear to users that toggling it on and off updates the theme. For example with latest breakage that was all that was required from users to have it working again. But this was not common knowledge so people freaked out.

Enabling it by default was my mistake. The rest of the gnome theme preferred vanilla by default, but I figured most people wouldn't discover it that way. Thing is, perhaps not everyone should have.

**Reply by rafaelmardojai** [reply]
Disabling it by default and improving the explanation sounds like a good start. Also maybe this repo can be linked to give the user more context and where they should report first any issue with the UI.

**Reply by Chrysostomus** [reply]
So instructions could be something like this? 

`If you encounter an ui issue on Firefox after enabling this theme: try disabling it and re-enabling it and restarting Firefox. This updates the theme to its latest version. If the issue persists, you can check the issue tracker in the theme repository <link to this repository>`

**Reply by emansom** [reply]
Most will glance over such statements and not remember it. Notify the user as well when either Firefox or this theme has been updated.

**Reply by Chrysostomus** [reply]
Not sure what would be the best way to implement that. We could add a pacman hook to trigger an update of theme when Firefox is updated, but the theme might not have yet caught up at that point, and in general, system packages should not touch files under $HOME. Having a script polling the upstream repo seems a bit excessive also, as many users want to control what sites their system contacts. 

What would be your suggested implementation of this?

**Reply by rafaelmardojai** [reply]
> `If you encounter an ui issue on Firefox after enabling this theme: try disabling it and re-enabling it and restarting Firefox. This updates the theme to its latest version. If the issue persists, you can check the issue tracker in the theme repository <link to this repository>`

Looks good. Also maybe you can ask for confirmation with an alert dialog when users enable the switch? I don't know if it will be too much, but a multi step process could help making things clear. I don't know how much prominent is this switch in your app, could you share some screenshots?

**Reply by Chrysostomus** [reply]
Confirmation dialog sounds like a good idea. That also solves the issue of where to place info text.

I'm on mobile, so I don't have a recent screenshot that has the current options, but the basic design looks like this:
![Screenshot from 2020-10-14 20-48-50](https://user-images.githubusercontent.com/12002226/208503474-cab35b3c-363d-4605-87fc-f1257498733c.jpg)

**Reply by rafaelmardojai** [reply]
Ok, so IMO a dialog with this contents works for me:

> **Enable Firefox GNOME theme**
>
> Be aware that this theme might do things that are not supported by upstream Firefox.
>
> If you encounter an UI issue on Firefox after enabling this theme: try disabling it and re-enabling it and restarting Firefox. This updates the theme to its latest version. If the issue persists, you can check the issue tracker in the theme repository \<link to this repository>.
>
> [Cancel] [I understand, continue]

**Reply by yochananmarqos** [reply]
@Ste74 likes to write little Yad GUI's like this. However, I think integrating it into Layouts itself would be best. @Chrysostomus @codesardine & are the Python guys, maybe they can implement it.

**Reply by rafaelmardojai** [reply]
Hello @Chrysostomus @yochananmarqos any update about the progress of this?

**Reply by Chrysostomus** [reply]
Not on the gui side, no. My schedule has been pretty hectic. But the theme is not installed by default anymore, so at least there is that for now. I'm hoping to get it done this month.

**Comment by yochananmarqos** [comment]
Current layout:

![image](https://user-images.githubusercontent.com/5437803/208503605-08f88ea3-e24e-4047-a39b-c44d21e71e2e.png)

**Comment by Chrysostomus** [comment]
![image](https://user-images.githubusercontent.com/12002226/236648231-fbfdadd5-b67f-46a0-9778-16ebd1a703a9.png)
Alright, this is the current implementation. Sorry for such a long delay.

**Reply by rafaelmardojai** [reply]
Looks good to have the info always visible, but I still would like some confirmation dialog when enabling.

**Reply by Chrysostomus** [reply]
Me too, mostly because it would aesthetically more pleasing. I've been pretty swamped with work and kids for months now, so this was the best I could manage as a stop gap measure until I have time to work on the dialog.

**Comment by rafaelmardojai** [comment]
Hi @Chrysostomus, now that GNOME has accent colors, I would like to remove the "themes" feature from the theme, would this affect you?

**Reply by Chrysostomus** [reply]
If the default theme follows the GTK/gnome color scheme, then dropping the custom style sheets would be ideal. Thank you for notification and consideration, I really appreciate it!

**Comment by yochananmarqos** [comment]
@rafaelmardojai I'm not sure what you're asking.

Either way, we now have a button that launches [Add Water.](https://github.com/largestgithubuseronearth/addwater).

![image](https://github.com/user-attachments/assets/b7ada1e9-80f4-4417-86e1-48b52be601e8)

**Reply by rafaelmardojai** [reply]
Hi @yochananmarqos , the theme ships with a `light-maia.css` and `dark-maia.css` stylesheets that can get enabled by the installation script, that styles basically just change the accent color to green. Now that GNOME has an accent color feature that works in Firefox 133, I want to get rid of those stylesheets (and the "themes" feature that loads them in the installation script), but i don't want to break anything for you.

**Reply by yochananmarqos** [reply]
Oh, I see. No, it would not affect anything as I've added Maia accent coloring elsewhere including Gtk3 applications like Firefox.
